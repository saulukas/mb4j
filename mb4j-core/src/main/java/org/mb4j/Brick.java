package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Mustache.Lambda;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class Brick implements Lambda {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderIndented(this, frag);
  }
}
