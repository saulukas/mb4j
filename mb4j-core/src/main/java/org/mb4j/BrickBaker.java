package org.mb4j;

public interface BrickBaker<P> {
  Brick bakeBrick(P params);
}
