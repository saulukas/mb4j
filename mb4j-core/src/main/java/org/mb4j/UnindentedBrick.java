package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;

public class UnindentedBrick extends Brick {
  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderUnindented(this, frag);
  }
}
