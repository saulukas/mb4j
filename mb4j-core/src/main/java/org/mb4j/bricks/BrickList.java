package org.mb4j.bricks;

import com.samskivert.mustache.Template.Fragment;
import java.io.Writer;
import org.mb4j.Brick;
import org.mb4j.renderer.RenderingScope;

public class BrickList extends Brick {

  public final Iterable<? extends Brick> items;

  public BrickList(Iterable<? extends Brick> items) {
    this.items = items;
  }

  @Override
  public void execute(Fragment frag, Writer out) {
    ((RenderingScope) out).renderList(items, frag);
  }

}
