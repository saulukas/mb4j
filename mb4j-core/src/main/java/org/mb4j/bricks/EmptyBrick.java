package org.mb4j.bricks;

import com.samskivert.mustache.Template;
import java.io.Writer;
import org.mb4j.Brick;

public class EmptyBrick extends Brick {
  private static final EmptyBrick INSTANCE = new EmptyBrick();

  public static EmptyBrick emptyBrick() {
    return INSTANCE;
  }

  @Override
  public void execute(Template.Fragment frag, Writer out) {
  }
}
