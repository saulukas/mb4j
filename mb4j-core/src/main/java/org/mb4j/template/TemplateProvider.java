package org.mb4j.template;

import org.mb4j.Brick;

public interface TemplateProvider {
  BrickTemplate templateFor(Class<? extends Brick> brickClass);
}
