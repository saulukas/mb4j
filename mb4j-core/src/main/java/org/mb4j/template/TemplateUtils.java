package org.mb4j.template;

import org.mb4j.Brick;

public class TemplateUtils {
  public static String typeStringOf(Class<? extends Brick> brickClass) {
    TemplateType annotation = brickClass.getAnnotation(TemplateType.class);
    return annotation == null ? TemplateType.DEFAULT : annotation.value();
  }

  public static String encodingStringOf(Class<? extends Brick> brickClass) {
    TemplateEncoding annotation = brickClass.getAnnotation(TemplateEncoding.class);
    return annotation == null ? TemplateEncoding.DEFAULT : annotation.value();
  }

  public static String outputEncodingStringOf(Class<? extends Brick> brickClass) {
    TemplateOutputEncoding annotation = brickClass.getAnnotation(TemplateOutputEncoding.class);
    return annotation == null ? TemplateOutputEncoding.DEFAULT : annotation.value();
  }
}
