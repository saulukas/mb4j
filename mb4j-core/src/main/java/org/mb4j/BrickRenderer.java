package org.mb4j;

import org.mb4j.renderer.RenderingScope;
import org.mb4j.template.TemplateProvider;
import java.io.Writer;

public class BrickRenderer {
  private final TemplateProvider templateProvider;

  public BrickRenderer(TemplateProvider templateProvider) {
    this.templateProvider = templateProvider;
  }

  public void render(Brick brick, Writer out) {
    RenderingScope scope = new RenderingScope(out, templateProvider);
    scope.render(brick);
  }
}
