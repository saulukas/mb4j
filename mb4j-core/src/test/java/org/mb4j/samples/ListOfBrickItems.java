package org.mb4j.samples;

import java.util.ArrayList;
import java.util.List;
import org.mb4j.Brick;
import org.mb4j.bricks.BrickList;
import org.mb4j.template.TemplateType;

@TemplateType(".mustache")
public class ListOfBrickItems extends Brick {

  BrickList verticalList;
  BrickList horizontalList;

  public ListOfBrickItems() {
    List<SimpleBrick> items = new ArrayList<SimpleBrick>();
    items.add(new SimpleBrick("water2"));
    items.add(new SimpleBrick("bread2"));
    items.add(new SimpleBrick("onion2"));
    this.verticalList = new BrickList(items);
    this.horizontalList = new BrickList(items);
  }

}
