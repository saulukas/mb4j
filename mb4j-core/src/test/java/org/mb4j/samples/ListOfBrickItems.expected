
    List of bricks:

    Just wanted to say water2 :)
    Just wanted to say bread2 :)
    Just wanted to say onion2 :)

    Just wanted to say water2 :) <---> Just wanted to say bread2 :) <---> Just wanted to say onion2 :)
